public/app/features/trails/utils.ts

```diff
@@ -115,3 +118,42 @@ export function getFilters(scene: SceneObject) {
   }
   return null;
 }
+
+// frontend hardening limit
+const MAX_ADHOC_VARIABLE_OPTIONS = 10000;
+
+export function limitAdhocProviders(
+  filtersVariable: SceneVariable<SceneVariableState> | null,
+  datasourceHelper: MetricDatasourceHelper
+) {
+  if (filtersVariable instanceof AdHocFiltersVariable) {
```

Comment:
Can we early return if filtersVariable is not an instance of AdHocFiltersVariable?
I think it would be easier to read the code

Reply:
Will return early!